### ISSUE TITLE
Latest firmware update to haiku fans broke compatibility 

### ISSUE THREAD

**Issue by pponce** [sean9keenan/homebridge-bigAssFans#40]
I believe the latest firmware updates to my haiku fans broke compatibility.
I suspect they changed the underlying api’s to work similar to how their newer i6 models work.
This is just a guess as they also migrated support for these fans to their newer app that supported i6 models. The Old bigassfan app no longer works with the updated haiku fans.

when I have time I’ll test the theory out by installing the i6 plugin linked to and mentioned here.
https://github.com/sean9keenan/homebridge-bigAssFans/issues/33

**Comment by jmissig** [comment]
Please keep us updated on what you find! Was really hoping they’d implement HomeKit software by now.

**Comment by pponce** [comment]
Me too. Wish they would add HomeKit support.
But I’m happy to report that the homebridge-i6-bigassfans plugin works!
https://github.com/oogje/homebridge-i6-bigAssFans

**Comment by pponce** [comment]
Some updates:
The i6 plugin mostly works.
The plugin does not poll for setting changes. So if you use the iOS bigassfan app to make changes or the ir remote then HomeKit won’t get the updates. Apparently the i6 fans push changes to HomeKit and the app when changes are made. Haiku fans don’t seem to push changes and require a pull from the plugin.

Also the temp is not recorded correctly in HomeKit. And i don’t think it has a humidity sensor even though one shows up in HomeKit for the haiku fans.

The plugin does do the main things though. Controls lights on and off with configurable %. Controls fan on and off with configurable %.

Hoping someone with more skillz than I can fork or update the i6 plugin to customize support for haiku fans.

**Comment by pponce** [comment]
The bigassfan iOS app does get status changes when I use the IR remote or the wifi wall controller. So the fan must be sending updates to the app somehow. So there is hope that updates can be pushed to HomeKit too.

**Comment by pponce** [comment]
update:

I've been collaborating with the developer of the [homebridge-i6-bigAssFans plugin](https://github.com/oogje/homebridge-i6-bigAssFans) and they were able to update the plugin to support getting updates to the status in homekit when IR remote or the App is used to make changes to the light or fan settings. He's also going to remove the humidity and dimToWarm features as those don't work with Haiku fans.
Install the beta like this:
npm install -g homebridge-i6-bigassfans@0.3.19-beta

**Comment by jmissig** [comment]
Any chance they'll get it auto-adding the fans instead of needing to add each one individually? Particularly since mine aren’t on fixed IP and I don’t intend to make them.

**Comment by pponce** [comment]
There is always a chance. I actually prefer specifying as it’s more reliable. But my hope is that there will be more users of the i6 plugin now that this update was pushed.  So we may get more community developer support for haiku fans.
The current owner of the project owns an i6 and does not own a haiku. That makes it a bit tough to work on Haiku specific support .  But we may get developers who own Haiku to provide PR’s for it.

I’m just super grateful we have the basics working and so fast. 
you should ask for the enhancement in the other project. See if it gets taken up.

And for what it’s worth. Mine are technically not on fixed ip either but the ip never changes for me.

**Comment by jmissig** [comment]
Oh it’s certainly great, I don’t mean to look a gift horse in the mouth! I just don’t want to have to be maintaining a list of IP addresses to MAC addresses. I’m mostly thinking about the threshold at which I update my Haiku’s firmware and go through with all this.

Yes it’ll just depend on your local DHCP server. I have enough IoT devices that they change sometimes.

**Comment by pponce** [comment]
Yup makes sense. I’ve not yet had to set aside ip addresses. I did like that with this, now not working plugin, you could specify number of fans to look for so it knows when to stop looking.

**Comment by buzz-id** [comment]
You won't imagine how much you've benefited my family and I! Thanks! Since the old Haiku Plug-ins. 

I'm new to GitHub but just checking if it works on the the H-Series.

**Comment by pponce** [comment]
@buzz-id thank the i6 plugin creator over here https://github.com/oogje/homebridge-i6-bigAssFans

He is working on a few more things before publishing his latest version. It should work on the H series as I believe that’s the type I have.

**Comment by buzz-id** [comment]
> @buzz-id thank the i6 plugin creator over here https://github.com/oogje/homebridge-i6-bigAssFans
> 
> 
> 
> He is working on a few more things before publishing his latest version. It should work on the H series as I believe that’s the type I have. 

Thanks! All hail the creator!
